File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (map.get(diff) != null) {
                int[] output = {map.get(diff), i};
                return output;
            } else {
                map.put(nums[i], i);
            }
        }
        return null;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        nums.sort(reverse=True)
        output = []
        def generator(index, target, generated_lst):
            if target == 0:
                output.append(generated_lst.copy())
                return
            if target < 0:
                return
            if index == len(nums):
                return

            # Pick the first number, then continue picking the first number
            chosen_num = nums[index]
            remaining = target - chosen_num
            generated_lst.append(chosen_num)
            generator(index, remaining, generated_lst)

            # Skip this number, then move on to pick the next number
            generated_lst.pop()
            generator(index + 1, target, generated_lst)

        generator(0, target, [])
        return output
